### ISSUE TITLE
Plugin callback is triggered every time submission step 4 is displayed

### ISSUE THREAD

**Issue by ewhanson** [asmecher/plagiarism#18]
### Description of the issue
The plugin callback that is responsible for submitting submission data to iThenticate is currently triggered by the `submissionsubmitstep4form::display` hook. This can result in the plugin callback being called multiple times for the same submission whenever the confirmatory submission step 4 is displayed. This may happen if the user refreshes the page or if they go back and edit previous steps and then return to step 4.

Duplicate submissions to iThenticate has come up a few times recently, first from a PKP PS client in December 2021 where a local customization was made to change this behaviour along with other minor changes; second via a report from Kathleen Luschek at Crossref, citing reports from users indicating they were also experiencing duplicate submissions to iThenticate.

### Proposed solution
I've tested a fix for this by changing the hook that triggers the plugin callback from `submissionsubmitstep4form::display` to `submissionsubmitstep4form::execute` so the submission is only sent to iThenticate once when the OJS submission process is finalized. @asmecher, does this sound like a reasonable approach? 

Kathleen also reported that she had heard from users that "their submissions are being duplicated, and these duplicates aren't being picked up by Turnitin because the plugin is creating a different folder for each submission." This approach doesn't change how anything to do with folders are handled and may be an unrelated issue.

**Comment by asmecher** [comment]
No objection to changing the callback from the step 4 display to the step 4 execute. See also https://github.com/asmecher/plagiarism/issues/13; a proper solution to this will have to wait until 3.4.x, when we'll have a proper queue setup for offline batch processing.

**Comment by ewhanson** [comment]
Thanks, that sounds good. I've created a PR, and once this is merged, can update the plugin gallery as well if you'd like.

PR: #19

**Comment by asmecher** [comment]
Have you tested it? Looks fine to me, but if you haven't, it would be good to give it a spin before deploying.

**Comment by ewhanson** [comment]
Yep, I've tested it locally, and it's working as expected (only triggering upon submission of the step 4 form).

**Comment by asmecher** [comment]
Great! Please go ahead, and an update to the Plugin Gallery would be welcome too.

**Comment by ewhanson** [comment]
I just remembered/realized I won't be able to make the release/merge the change. Would you be able to merge #19? After that I'll follow up with an update to the plugin gallery repo. Thanks!

**Comment by asmecher** [comment]
Of course, thanks!
